File: YLWebViewForH5/YLWebViewForH5/ViewController.m
//
//  ViewController.m
//  YLWebViewForH5
//
//  Created by 王留根 on 2018/7/9.
//  Copyright © 2018年 OwnersVote. All rights reserved.
//

#import "ViewController.h"
#import "YLWKWebView.h"

@interface ViewController ()<WKNavigationDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    //   let strUrl = "http://192.168.20.14:8080/protocal"/index.html
    NSString * strUrl = @"http://192.168.20.14:8080/index.html";
    YLWKWebView *webView = [[YLWKWebView alloc]initWithFrame:CGRectMake(0, 0, [UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height - 64)];
    webView.backgroundColor = [UIColor clearColor];
    webView.progressCorlor = [UIColor greenColor];
    webView.navigationDelegate = self;
    [self.view addSubview:webView];
    
    NSURLRequest *request =[NSURLRequest requestWithURL:[NSURL URLWithString:[strUrl stringByAddingPercentEncodingWithAllowedCharacters: [NSCharacterSet URLQueryAllowedCharacterSet]]]];
    [webView loadRequest:request];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: YLSwiftProtocolTest/YLSwiftProtocolTest/ViewController.swift
//
//  ViewController.swift
//  YLSwiftProtocolTest
//
//  Created by 王留根 on 17/2/24.
//  Copyright © 2017年 ios-mac. All rights reserved.
//

import UIKit
import YLSwiftScan

protocol TestProtocol: class {
    func needAchieve() ->String
    func notNeedAchieve() ->String
}

protocol putInProtocol: class {
    func input(name: String) -> Void
}
protocol putOutProtocol: class {
    func output(name: String) -> Void
}

extension TestProtocol {
    func notNeedAchieve() ->String {
        return " 不需要实现"
    }
   
}

class ViewController: UIViewController {

     weak var delegate: putInProtocol?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        let templet: ObjectTemple = ObjectTemple();
        templet.delegate = self;
        self.delegate = templet as putInProtocol;
        if let delegate = self.delegate {
            delegate.input(name: "这就是命啊");
        }
        print("\(needAchieve())")
        print("\(notNeedAchieve())")
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

extension ViewController: putOutProtocol {
    func output(name: String) {
        print("name : \(name)");
    }
}

extension ViewController: TestProtocol {
    
    func needAchieve() -> String {
        return "必学实现"
    }
//    func notNeedAchieve() -> String {
//        return "****"
//    }
    
}

